File: ROS/pythonNodes/number_subscriber.py

''''

Author: Mahshad Nazari
Date: 06/03/2022
Topic: ROS workshop
Name: First Topic (Subscriber)

'''

import rospy
from std_msgs.msg import Int64


def callback_number(msg):
	rospy.loginfo(msg.data)


if __name__ == '__main__':
	rospy.init_node('number_subscriber')
	sub = rospy.Subscriber("/number", Int64, callback_number)
	rospy.spin()

File: ROS/cppNodes/number_subscriber.cpp
/*

Author: Mahshad Nazari
Date: 06/03/2022
Name: First C++ Subscriber

*/

#include <ros/ros.h>
#include <std_msgs/Int64.h>

int counter = 0;
ros::Publisher pub;

void callback_number(const std_msgs::Int64& msg)
{
	ROS_INFO("%d", msg.data);
}

int main (int argc, char **argv)
{
	ros::init(argc, argv, "number_counter");
	ros::NodeHandle nh;

	ros::Subscriber sub = nh.subscribe("/number", 1000, callback_number);

	ros::spin();
}
